bookShop/src/app/state/collection.reducer.ts

```diff
@@ -0,0 +1,15 @@
+import { createReducer, on } from '@ngrx/store';
+import { IBook } from '../shared/models/BookModel';
+import { addBook, removeBook } from './books.actions';
+import { retrievedBookList } from './books.actions';
+
+export const initialState: ReadonlyArray<IBook> = [];
```

Comment:
Let's make a store like an object, not an array.

Reply:
For what?

Reply:
Fixed